python/lsst/ip/isr/isrTask.py

```diff
@@ -574,16 +713,117 @@ class IsrTask(pipeBase.CmdLineTask):
+            del inputTypeDict["atmosphereTransmission"]
+
+        return inputTypeDict
+
+    @classmethod
+    def getOutputDatasetTypes(cls, config):
+        outputTypeDict = super().getOutputDatasetTypes(config)
+        obsConfigDir = os.path.join(getPackageDir("obs_subaru"), "config", "hsc")
+        config.load(os.path.join(obsConfigDir, "isr.py"))
+
+        if config.qa.doThumbnailOss is not True:
+            del outputTypeDict["outputOssThumbnail"]
+        if config.qa.doThumbnailFlattened is not True:
+            del outputTypeDict["outputFlattenedThumbnail"]
+
+        return outputTypeDict
+
+    def adaptArgsAndRun(self, inputData, inputDataIds, outputDataIds, butler):
+        try:
+            inputData['detectorNum'] = int(inputDataIds['ccdExposure']['detector'])
+        except Exception as e:
+            raise ValueError("Failure to find valid detectorNum value for Dataset {}: {}".format(
+                inputDataIds, e))
+
+        inputData['isGen3'] = True
+
+        if self.config.doLinearize is True:
+            if 'linearizer' not in inputData.keys():
+                detector = inputData['camera'][inputData['detectorNum']]
+                linearityName = detector.getAmpInfoCatalog()[0].getLinearityType()
+                inputData['linearizer'] = linearize.getLinearityTypeByName(linearityName)()
+
+        if inputData['defects'] is not None:
+            # defects is loaded as a BaseCatalog with columns x0, y0, width, height.
+            # masking expects a list of defects defined by their bounding box
+            defectList = []
+
+            for r in inputData['defects']:
+                bbox = afwGeom.BoxI(afwGeom.PointI(r.get("x0"), r.get("y0")),
+                                    afwGeom.ExtentI(r.get("width"), r.get("height")))
+                defectList.append(Defect(bbox))
+
+            inputData['defects'] = defectList
+
+        # Broken: DM-17169
+        # ci_hsc does not use crosstalkSources, as it's intra-CCD CT only.  This needs to be
+        # fixed for non-HSC cameras in the future.
+        # inputData['crosstalkSources'] = (self.crosstalk.prepCrosstalk(inputDataIds['ccdExposure'])
+        #                        if self.config.doCrosstalk else None)
+
+        # Broken: DM-17152
+        # Fringes are not handled correctly by Gen3 butler.  We need to be able to return the
+        # fringe if the filter contains one, but not fail if the filter doesn't.
+        # inputData['fringes'] = (self.fringe.readFringes(inputDataIds['ccdExposure'],
+        #                                                 assembler=self.assembleCcd
+        #                                                 if self.config.doAssembleIsrExposures else None)
+        #                         if self.config.doFringe and
+        #                            self.fringe.checkFilter(inputData['ccdExposure'])
+        #                         else pipeBase.Struct(fringes=None))
```

Comment:
So you are saying there is some HSC specific way to do this in `obs_subaru` that will work for `ci_hsc`, but for now that won't work for any other cameras?

Reply:
I've changed the comment a bi.  Gen3 is currently only testable with `ci_hsc`, but `ci_hsc` has no y-band data to test that fringes load correctly.  A block like that used for the defects may be needed, or this all will get wrapped into a new butler formatter that will handle repacking data structures.